rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java

```diff
@@ -288,12 +290,14 @@ public void add(Channel peer) {
     public void notifyDisconnect(Channel channel) {
         logger.debug("Peer {}: notifies about disconnect", channel.getPeerIdShort());
         channel.onDisconnect();
-        syncPool.onDisconnect(channel);
-        synchronized (activePeersLock){
-            activePeers.values().remove(channel);
-        }
-        if(newPeers.remove(channel)) {
-            logger.info("Peer removed from active peers: {}", channel.getPeerId());
+        synchronized (newPeers) {
```

Comment:
Why not add synchronization to add(Channel peer)?

Reply:
It wouldn't make a difference. The newPeers array itself is thread safe. EthereumChannelInitializer both calls this add method and registers the listener that will call notifyDisconnect, and it does the add first, so there is no possible race condition there.

See https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java#L119